apps/src/code-studio/components/progress/TeacherUnitOverview.tsx

```diff
@@ -239,6 +240,16 @@ const initializeRedux = (
     dispatch(setStudentDefaultsSummaryView(false));
   }
 
+  if (unitData.showCalendar) {
```

Comment:
We need to setShowCalendar to false if the new unit does not show the calendar. Otherwise, I think we could start on a unit that shows the calendar and go to a unit that does not have the calendar and still have this set to true.

Reply:
This might actually be a problem with the above if statements as well. Would you mind taking a look to see if my test case above is happening and if it is, could you fix the other if statements?